langchain/document_loaders/pdf.py

```diff
@@ -381,3 +381,91 @@ def load(self) -> List[Document]:
             contents = self.clean_pdf(contents)
         metadata = {"source": self.source, "file_path": self.source}
         return [Document(page_content=contents, metadata=metadata)]
+
+
+class PDFPlumberLoader(BasePDFLoader):
+    """Loader that uses PDFPlumber to load PDF files."""
+
+    def __init__(self, file_path: str):
+        """Initialize with file path."""
+        try:
+            import pdfplumber  # noqa:F401
+        except ImportError:
+            raise ValueError(
+                "PDFPlumber package not found, please install it with "
+                "`pip install pdfplumber`"
+            )
+
+        super().__init__(file_path)
+
+    def load(self, **kwargs: Optional[Any]) -> List[Document]:
+        """Load file."""
+        import pdfplumber
+
+        doc = pdfplumber.open(self.file_path)
+        file_path = self.file_path if self.web_path is None else self.web_path
```

Comment:
think this is same as `self.source`, could just use that

Reply:
Fixed. I also noticed the if-else statement in PyMuPDFLoader, I think those should be fixed as well?